plugins/woocommerce/includes/wc-product-functions.php

```diff
@@ -158,6 +173,67 @@ function wc_delete_product_transients( $post_id = 0 ) {
 	do_action( 'woocommerce_delete_product_transients', $post_id );
 }
 
+/**
+ * Asynchronously delete related product transients when a product is changed.
+ * This is necessary because changing one product might affect many related products.
+ *
+ * @since 9.8.1
+ * @param array $post_id Arguments passed from the async scheduler.
+ */
+function wc_delete_related_product_transients( $post_id ) {
+	global $wpdb;
+
+	if ( ! is_numeric( $post_id ) ) {
+		return;
+	}
+
+	$transient_name       = 'wc_related_' . $post_id;
+	$old_transient        = get_transient( $transient_name );
+	$old_related_products = array();
+
+	if ( is_array( $old_transient ) ) {
```

Comment:
Lets also check it's not an empty array

```suggestion
	if ( is_array( $old_transient ) && ! empty( $old_transient ) ) {
```